widget/textgrid.go

```diff
@@ -47,9 +47,19 @@ type TextGridStyle interface {
 	BackgroundColor() color.Color
 }
 
+// TextGridTextStyle defines the text style that can be applied to a TextGrid cell.
+//
+// Since: 2.5
+type TextGridTextStyle interface {
+	TextGridStyle
+	Bold() bool
+	Underlined() bool
```

Comment:
Isn't it `Underline` not `Underlined` in most places?

Reply:
Underline only appears in the processing logic to refer to the actual line.

Reply:
Sorry, I was meaning in most typographic APIs is it not referred to as underline?

I.e. it is not "bolded" or "italicised" either...

Reply:
These methods act as queries, determining whether a specific style is currently applied. "IsUnderline" wouldn't be appropriate because it suggests an action to check for underlining, rather than reflecting the current state of the text.
if Bold() accurately reflects the presence of bold formatting, then "Underlined()" follows the same logic. It's not "IsBolded" because the text either is bolded or not.
While "underline" might be more common in typography, "Underlined()" aligns with the existing naming convention and emphasizes the query nature of these methods.  This prioritizes readability and consistency within the codebase.

Reply:
I don't see how this is consistent, "Bold" and "Italic" used elsewhere refer to state. Their "is set" equivalent in getters would be "IsItalic" or "Italicised".
Following this "Underline" is the state version and underlined refers to an action that occurred.

As said in other places though, I'm still not sure why it isn't a better plan to use existing "TextStyle" API rather than a specific style API for this widget.

Reply:
This is a strange discussion but I'm going to weigh in anyway.

```
You would say:      "Is it bold?"
You would not say:  "Is it bolded?"

You would say:      "Is it italic?"
You would not say:  "Is it italicised?"

You would say:      "Is it underlined?"
You would not say:  "Is it underline?"
```

```go
if text.Bold() { log.Debug("Text is bold") }

if text.Italic() { log.Debug("Text is italic") }

if text.Underlined() { log.Debug("Text is underlined") }
```

Put another way:

"Italic" is a adjective/noun, it describes what something is, its a name or state of something.
"Underline" (in how it's used here) is a verb, it's an action, it is not a description or name and doesn't represent state.

If we used your version the code would be:

```go
if text.Underline() { log.Debug("Text is underline") }
```

Which reads and sounds (especially if you say it out loud) very wrong.

Reply:
> As said in other places though, I'm still not sure why it isn't a better plan to use existing "TextStyle" API rather than a specific style API for this widget.

This would make sense, having different versions of effectively the same thing isn't great, assuming there's no gotchas lurking under the code for why this wouldn't work.

Reply:
To weigh in a bit more, I think the confusion is coming from the fact this is a method on a struct and not an enum like more traditional API's.
Since this is a method it feels more subject to OO programming naming, like we have suggested. If this was an enum then I suspect this whole thing would go away.
some other GUI API's for reference. 
https://learn.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-createfonta?redirectedfrom=MSDN
https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/javax/swing/text/StyleConstants.CharacterConstants.html

Reply:
>>As said in other places though, I'm still not sure why it isn't a better plan to use existing "TextStyle" API rather than a specific style API for this widget.

>This would make sense, having different versions of effectively the same thing isn't great, assuming there's no gotchas lurking under the code for why this wouldn't work.

The only gotchas would be that I'm only implementing this for this single widget, all other widgets can be updated subsequently by the parties wanting that functionality.

Reply:
> > This would make sense, having different versions of effectively the same thing isn't great, assuming there's no gotchas lurking under the code for why this wouldn't work.
> 
> The only gotchas would be that I'm only implementing this for this single widget, all other widgets can be updated subsequently by the parties wanting that functionality.

I think it's reasonable to use the existing TextStyle then, add the new features, support them in the widget you need it for and anyone who wants to add support to the other widgets can if/when it's needed.

Reply:
Ah, difference between field and method yes this is causing confusion.

> The only gotchas would be that I'm only implementing this for this single widget, all other widgets can be updated subsequently by the parties wanting that functionality.

We should not add temporary APIs or items that are widget specific but will be broadened. Any public API will remain part of the toolkit indefinitely. Given there are feature requests open for underline it seems pretty reasonable to think it will be added in the meantime we could add some documentation about it not being supported in all locations until that is no longer true.